managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpScan.java

```diff
@@ -128,7 +128,7 @@ public void find() {
         }
         if (cursor.hasMoreEntries(searchPosition)) {
             OpReadEntry opReadEntry = OpReadEntry.create(cursor, searchPosition, batchSize,
-            this, OpScan.this.ctx, null, null);
+            this, OpScan.this.ctx, null, null, false);
```

Comment:
Instead of adding more complexity to the original entry reads. I think we can have a separate non-durable cursor to do the scanning. The original cursor (users' subscription) just provided a start position and the entry filter of the scanning task.

And it will provide good observability for the scanning task. You can get the non-durable cursor stats from the topic's internal-stats.

Reply:
Done